Spring Boot compatibility differs depending on type of usage

<!--- Thank you for taking the time to create a report to help us improve -->

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
<!--- Tell us what should happen -->
When using CM4SB 2.3.0 I would expect the version of Spring Boot that it can be used with to be the same whether CM4SB is a build-time dependency of the Boot project (i.e. a Gradle/Maven dependency) or else an external runtime dependency (used against an existing Spring Boot executable Jar with the PropertiesLauncher). 

## Current Behavior
<!--- Tell us what happens instead of the expected behavior -->
When used as a build-time dependency, CM4SB appears to work well with Spring Boot 2.3.8 and 2.4.2 applications.

When the PropertiesLauncher approach is used (incorporating **chaos-monkey-spring-boot-2.3.0-jar-with-dependencies.jar**) with a simple Spring Boot 2.3.8 web application then everything starts up fine but when the first HTTP request is sent to it a `java.lang.NoSuchMethodError` is thrown...
```bash
Caused by: java.lang.NoSuchMethodError: 'boolean org.springframework.util.StringUtils.matchesCharacter(java.lang.String, char)'
	at org.springframework.web.util.UrlPathHelper.getContextPath(UrlPathHelper.java:398) ~[spring-web-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.web.util.UrlPathHelper.getPathWithinApplication(UrlPathHelper.java:297) ~[spring-web-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.web.util.UrlPathHelper.getLookupPathForRequest(UrlPathHelper.java:186) ~[spring-web-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.web.servlet.handler.AbstractHandlerMethodMapping.getHandlerInternal(AbstractHandlerMethodMapping.java:363) ~[spring-webmvc-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.web.servlet.mvc.method.RequestMappingInfoHandlerMapping.getHandlerInternal(RequestMappingInfoHandlerMapping.java:110) ~[spring-webmvc-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.web.servlet.mvc.method.RequestMappingInfoHandlerMapping.getHandlerInternal(RequestMappingInfoHandlerMapping.java:59) ~[spring-webmvc-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.web.servlet.handler.AbstractHandlerMapping.getHandler(AbstractHandlerMapping.java:396) ~[spring-webmvc-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.getHandler(DispatcherServlet.java:1234) ~[spring-webmvc-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1016) ~[spring-webmvc-5.2.12.RELEASE.jar!/:5.2.12.RELEASE]
	... 38 common frames omitted
``` 
The "problem" method is `org.springframework.util.StringUtils.matchesCharacter(java.lang.String, char)` which was introduced to Spring in version 5.2.9.

The identical error occurs for the following versions of Spring Boot:
- 2.3.7 (which pulls in Spring 5.2.12)
- 2.3.6 (Spring 5.2.11)
- 2.3.5 (Spring 5.2.10)
- 2.3.4 (Spring 5.2.9)

Things only begin to work healthily when Spring Boot 2.3.3 is used. This is based off Spring 5.2.8 which does not include the `org.springframework.util.StringUtils.matchesCharacter(java.lang.String, char)` method.

## Possible Solution
<!--- Not obligatory, if possible suggest a fix/reason for the bug, -->
Peering at the contents of **chaos-monkey-spring-boot-2.3.0-jar-with-dependencies.jar** it contains its own copy of the `org.springframework.util.StringUtils` class dated Tuesday March 24 2020. My assumption is that this older version of the class is being loaded before the one contained by the Spring Boot application we want to inject chaos into. This results in the referenced method not being found at runtime when Spring Boot is above 2.3.3 (Spring Core 5.2.8). 

Refreshing the "with-dependencies" Jar with more recent Spring core classes should probably help in this matter.

--- a/.mvn/wrapper/maven-wrapper.properties
+++ b/.mvn/wrapper/maven-wrapper.properties
@@ -1,2 +1,2 @@
-distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.6.2/apache-maven-3.6.2-bin.zip
-wrapperUrl=https://repo.maven.apache.org/maven2/io/takari/maven-wrapper/0.5.5/maven-wrapper-0.5.5.jar
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.6.3/apache-maven-3.6.3-bin.zip
+wrapperUrl=https://repo.maven.apache.org/maven2/io/takari/maven-wrapper/0.5.6/maven-wrapper-0.5.6.jar

--- a/CONTRIBUTING.adoc
+++ b/CONTRIBUTING.adoc
@@ -77,8 +77,7 @@ press the gear button and choose `Import Scheme`. Then, select the downloaded th
 == Working With The Code
 Considering these recommendations will help us in getting your pull request merged. 
 
-* Add the ASF license header comment to all new `.java` files (you can copy from existing files in the project). 
-* Add yourself as an `@author` to the `.java` files you modify.
+* Add the ASF license header comment to all new `.java` files (you can copy from existing files in the project).
 * Document your changes. This includes in code documentation but also user documentation.
 * Test your changes. Use `./mvnw verify` to run all build pipeline checks.
 * It would be great to follow Chris Beam's recommendations on https://chris.beams.io/posts/git-commit/[How to Write a Git Commit Message]. 

--- a/chaos-monkey-docs/src/main/asciidoc/changes.adoc
+++ b/chaos-monkey-docs/src/main/asciidoc/changes.adoc
@@ -9,6 +9,7 @@
 === Improvements
 // - https://github.com/codecentric/chaos-monkey-spring-boot/pull/xxx[#xxx] Added example entry. Please don't remove.
 - https://github.com/codecentric/chaos-monkey-spring-boot/pull/204[#204] [CI] Added release workflow
+- https://github.com/codecentric/chaos-monkey-spring-boot/pull/213[#213] Upgrade to Spring Boot 2.3.9
 
 === New Features
 // - https://github.com/codecentric/chaos-monkey-spring-boot/pull/xxx[#xxx] Added example entry. Please don't remove.

--- a/demo-apps/chaos-monkey-demo-app-ext-jar/pom.xml
+++ b/demo-apps/chaos-monkey-demo-app-ext-jar/pom.xml
@@ -44,7 +44,7 @@
         <dependency>
             <groupId>io.micrometer</groupId>
             <artifactId>micrometer-registry-prometheus</artifactId>
-            <version>1.5.7</version>
+            <version>1.5.11</version>
         </dependency>
         <dependency>
             <groupId>org.springframework.boot</groupId>

--- a/demo-apps/chaos-monkey-demo-app/pom.xml
+++ b/demo-apps/chaos-monkey-demo-app/pom.xml
@@ -60,7 +60,7 @@
         <dependency>
             <groupId>io.micrometer</groupId>
             <artifactId>micrometer-registry-prometheus</artifactId>
-            <version>1.5.1</version>
+            <version>1.5.11</version>
         </dependency>
         <dependency>
             <groupId>org.springframework.boot</groupId>

--- a/demo-apps/chaos-monkey-web-reactive-app/pom.xml
+++ b/demo-apps/chaos-monkey-web-reactive-app/pom.xml
@@ -48,7 +48,7 @@
         <dependency>
             <groupId>io.micrometer</groupId>
             <artifactId>micrometer-registry-prometheus</artifactId>
-            <version>1.5.7</version>
+            <version>1.5.11</version>
         </dependency>
         <dependency>
             <groupId>org.springframework.boot</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -22,47 +22,46 @@
 
   <properties>
     <!-- project version -->
-    <asciidoctor-maven-plugin.version>1.6.0</asciidoctor-maven-plugin.version>
+    <revision>2.3.2-SNAPSHOT</revision>
     <!-- default settings -->
-    <aspectj-weaver.version>1.9.6</aspectj-weaver.version>
-    <cobertura-maven-plugin.version>2.7</cobertura-maven-plugin.version>
-    <com.puppycrawl.tools.version>8.38</com.puppycrawl.tools.version>
-    <flatten-maven-plugin.version>1.2.2</flatten-maven-plugin.version>
-    <!-- dependency versions -->
-    <git-commit-id-plugin.version>4.0.3</git-commit-id-plugin.version>
-    <github.global.server>github</github.global.server>
-    <jacoco-maven-plugin.version>0.8.6</jacoco-maven-plugin.version>
     <java.version>1.8</java.version>
-    <jcl-over-slf4j.version>1.7.30</jcl-over-slf4j.version>
+    <maven.compiler.source>${java.version}</maven.compiler.source>
+    <maven.compiler.target>${java.version}</maven.compiler.target>
+
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+
+    <!-- GitHub Server settings.xml -->
+    <github.global.server>github</github.global.server>
+
+    <!-- spring versions -->
+    <spring-boot.version>2.3.9.RELEASE</spring-boot.version>
+    <spring-cloud.version>Hoxton.SR10</spring-cloud.version>
+    <spring.version>5.2.13.RELEASE</spring.version>
+
+    <!-- dependency versions -->
     <junit.jupiter.version>5.7.0</junit.jupiter.version>
+    <mockito.version>3.6.28</mockito.version>
+    <lombok.version>1.18.18</lombok.version>
+    <aspectj-weaver.version>1.9.6</aspectj-weaver.version>
+    <jcl-over-slf4j.version>1.7.30</jcl-over-slf4j.version>
+
     <!-- plugin versions-->
-    <lombok.version>1.18.16</lombok.version>
+    <jacoco-maven-plugin.version>0.8.6</jacoco-maven-plugin.version>
+    <cobertura-maven-plugin.version>2.7</cobertura-maven-plugin.version>
+    <git-commit-id-plugin.version>4.0.3</git-commit-id-plugin.version>
+    <asciidoctor-maven-plugin.version>1.6.0</asciidoctor-maven-plugin.version>
     <maven-deploy-plugin.version>3.0.0-M1</maven-deploy-plugin.version>
     <maven-gpg-plugin.version>1.6</maven-gpg-plugin.version>
     <maven-javadoc-plugin.version>3.2.0</maven-javadoc-plugin.version>
     <maven-resources-plugin.version>3.2.0</maven-resources-plugin.version>
     <maven-source-plugin.version>3.2.0</maven-source-plugin.version>
     <maven-surefire-plugin.version>3.0.0-M5</maven-surefire-plugin.version>
-    <maven.compiler.source>${java.version}</maven.compiler.source>
-
-    <!-- GitHub Server settings.xml -->
-    <maven.compiler.target>${java.version}</maven.compiler.target>
-
-    <mockito.version>3.6.28</mockito.version>
-    <nexus-staging-maven-plugin.version>1.6.8
-    </nexus-staging-maven-plugin.version>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <project.reporting.outputEncoding>UTF-8
-    </project.reporting.outputEncoding>
-    <revision>2.3.2-SNAPSHOT</revision>
-    <site-maven-plugin.version>0.12</site-maven-plugin.version>
-    <spring-boot.version>2.3.6.RELEASE</spring-boot.version>
-    <spring-cloud.version>Hoxton.SR9</spring-cloud.version>
-
-    <spring.version>5.2.11.RELEASE</spring.version>
-
-    <google-java-format.version>1.7</google-java-format.version>
+    <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>
+    <flatten-maven-plugin.version>1.2.2</flatten-maven-plugin.version>
     <spotless.plugin.version>2.6.1</spotless.plugin.version>
+    <google-java-format.version>1.7</google-java-format.version>
+    <site-maven-plugin.version>0.12</site-maven-plugin.version>
   </properties>
 
   <scm>

